File: final_dataset_A/generated_code/claude/java/Post_blog_comment_021_Vulnerable.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class Post_blog_comment_021_Vulnerable {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String postComment(String name, String email, String comment) {
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {

            String query = "INSERT INTO blog_comments (name, email, comment) " +
                    "VALUES ('" + name + "', '" + email + "', '" + comment + "')";
            stmt.executeUpdate(query);
            return "Comment posted successfully";
        } catch (SQLException e) {
            return "Failed to post comment: " + e.getMessage();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Post_blog_comment_021_Vulnerable.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def post_comment(name, email, comment):
    """Let users post comments on a blog article."""
    conn = get_connection()
    cursor = conn.cursor()

    query = "INSERT INTO blog_comments (name, email, comment) " \
            "VALUES ('%s', '%s', '%s')" % (name, email, comment)
    try:
        cursor.execute(query)
        conn.commit()
        return "Comment posted successfully"
    except Exception as e:
        return "Failed to post comment: " + str(e)
    finally:
        conn.close()
